backend/src/models/reports/Tab_49_rpt_PF_NetRecoveries.js

```diff
@@ -2,13 +2,202 @@ const dbConnection = require("@database/databaseConnection");
 const { knex } = dbConnection();
 
 /**
- * Gets data for the Divisional Project Reports - Project Dashboard report.
+ * Gets data for the Divisional Project Reports - GDX Project Net Recoveries report.
  *
+ * @param           requestParams fiscal: the fiscal year for this report portfolio_id: the portfolio for this report
  * @returns {any[]}
  */
-const Tab_49_rpt_PF_NetRecoveries = () => {
-  const query = knex().raw();
 
+const handleParams = (query, requestParams) => {
+  if (requestParams.fiscal) {
+    query.where({
+      "q.fiscal": requestParams.fiscal,
+    });
+  }
+};
+
+const Tab_49_rpt_PF_NetRecoveries = (requestParams) => {
+  const query = knex()
+    .select("*")
+    .fromRaw(
+      `(WITH current_year_recoveries_stob_base AS (
```

Comment:
this might be cleaner, not that it matters
```
WITH current_year_recoveries_stob_base AS (
  SELECT
    p.id AS project_id,
    p.project_number,
    p.project_name,
    p.recoverable,
    po.id AS portfolio_id,
    po.portfolio_name,
    po.portfolio_abbrev,
    p.total_project_budget,
    p.recoverable_amount,
    pb.stob,
    pb.q1_recovered,
    pb.q1_amount,
    pb.q2_recovered,
    pb.q2_amount,
    pb.q3_recovered,
    pb.q3_amount,
    pb.q4_recovered,
    pb.q4_amount,
    fy.fiscal_year,
    pd.fiscal
  FROM
    data.project_budget AS pb
    LEFT JOIN data.project_deliverable AS pd ON pb.project_deliverable_id = pd.id
    LEFT JOIN data.project AS p ON pd.project_id = p.id
    LEFT JOIN data.fiscal_year AS fy ON pd.fiscal = fy.id
    LEFT JOIN data.portfolio AS po ON pb.recovery_area = po.id
),
current_year_recoveries_stob_no_param AS (
  SELECT
    c.project_id,
    c.project_number,
    c.project_name,
    c.recoverable,
    c.portfolio_id,
    c.portfolio_name,
    c.portfolio_abbrev,
    c.total_project_budget,
    c.recoverable_amount,
    c.stob,
    SUM(
      c.q1_amount + c.q2_amount + c.q3_amount + c.q4_amount
    ) AS current_fy_total_recoverable,
    SUM(
      CASE
        WHEN c.q1_recovered THEN c.q1_amount
        ELSE 0::money
      END +
      CASE
        WHEN c.q2_recovered THEN c.q2_amount
        ELSE 0::money
      END +
      CASE
        WHEN c.q3_recovered THEN c.q3_amount
        ELSE 0::money
      END +
      CASE
        WHEN c.q4_recovered THEN c.q4_amount
        ELSE 0::money
      END
    ) AS current_fy_recovered_to_date,
    c.fiscal_year,
    c.fiscal
  FROM
    current_year_recoveries_stob_base AS c
  GROUP BY
    c.project_id,
    c.project_number,
    c.project_name,
    c.recoverable,
    c.portfolio_id,
    c.portfolio_name,
    c.portfolio_abbrev,
    c.total_project_budget,
    c.recoverable_amount,
    c.stob,
    c.fiscal_year,
    c.fiscal
),
financial_recovery_by_portfolio_and_stob_recovered_no_param AS (
  SELECT
    c.project_id,
    c.portfolio_id,
    c.fiscal_year,
    SUM(
      c.current_fy_recovered_to_date
    ) FILTER (
      WHERE c.stob IN ('6309', '6310', '6001', '6002', '6398')
    ) AS contract_costs,
    SUM(
      c.current_fy_recovered_to_date
    ) FILTER (
      WHERE c.stob LIKE '57%'
    ) AS travel_costs,
    SUM(
      c.current_fy_recovered_to_date
    ) FILTER (
      WHERE c.stob LIKE '65%'
    ) AS business_expenses,
    SUM(
      c.current_fy_recovered_to_date
    ) FILTER (
      WHERE c.stob LIKE '88%'
    ) AS staff_recoveries,
    SUM(
      c.current_fy_recovered_to_date
    ) FILTER (
      WHERE (
        c.stob NOT IN ('6309', '6310', '6001', '6002', '6398')
      )
      AND (
        c.stob NOT LIKE '57%'
      )
      AND (
        c.stob NOT LIKE '65%'
      )
      AND (
        c.stob NOT LIKE '88%'
      )
    ) AS other
  FROM
    current_year_recoveries_stob_no_param AS c
  GROUP BY
    c.project_id,
    c.portfolio_id,
    c.fiscal_year,
    CASE
      WHEN c.stob IN ('6309', '6310', '6001', '6002', '6398') THEN 'contract_costs'
      WHEN c.stob LIKE '57%' THEN 'travel_costs'
      WHEN c.stob LIKE '65%' THEN 'business_expenses'
      WHEN c.stob LIKE '88%' THEN 'staff_recoveries'
      ELSE 'other'
    END
)
SELECT
  SUM(c.current_fy_total_recoverable) AS sum_of_current_fy_total_recoverable,
  c.project_id,
  c.project_number,
  c.project_name,
  c.portfolio_id,
  c.portfolio_name,
  c.total_project_budget,
  c.fiscal_year AS fy,
  c.fiscal,
  f.contract_costs AS contract_recovered,
  f.travel_costs AS travel_recovered,
  f.business_expenses AS business_expense_recovered,
  f.staff_recoveries AS staff_recoveries_recovered,
  f.other AS other_recovered,
  SUM(c.current_fy_recovered_to_date) AS cy_recovered_to_date,
  SUM(c.current_fy_total_recoverable) AS cy_total_recoverable
FROM
  financial_recovery_by_portfolio_and_stob_recovered_no_param AS f
  INNER JOIN current_year_recoveries_stob_no_param AS c ON (
    f.fiscal_year = c.fiscal_year
  )
  AND (
    f.portfolio_id = c.portfolio_id
  )
  AND (
    f.project_id = c.project_id
  )
GROUP BY
  c.project_id,
  c.project_number,
  c.project_name,
  c.portfolio_id,
  c.portfolio_name,
  c.portfolio_abbrev,
  c.total_project_budget,
  c.total_project_budget - c.recoverable_amount,
  c.fiscal_year,
  c.fiscal,
  f.contract_costs,
  f.travel_costs,
  f.business_expenses,
  f.staff_recoveries,
  f.other
ORDER BY
  c.portfolio_name,
  c.project_number;
```